src/SingleDatePicker.elm

```diff
@@ -180,40 +205,32 @@ clickedOutsidePicker settings componentId internalMsg datePicker =
             )
 
 
-calculatePickerOffset : Zone -> Posix -> Maybe Posix -> Int
-calculatePickerOffset zone baseTime pickedTime =
-    let
-        flooredBase =
-            Time.floor Month zone baseTime
-    in
-    case pickedTime of
-        Nothing ->
-            0
-
-        Just time ->
-            let
-                flooredPick =
-                    Time.floor Month zone time
-            in
-            if Time.posixToMillis flooredBase <= Time.posixToMillis flooredPick then
-                Time.diff Month zone flooredBase flooredPick
-
-            else
-                0 - Time.diff Month zone flooredPick flooredBase
-
-
 {-| Open the provided date picker and receive the updated picker instance. Also
 takes a default time the picker should center on (in the event a time has not yet
 been picked) as well as the picked time. A common example of a default time
 would be the datetime for the current day.
 -}
-openPicker : Zone -> Posix -> Maybe Posix -> DatePicker -> DatePicker
-openPicker zone baseTime pickedTime (DatePicker model) =
+openPicker : Settings msg -> Posix -> Maybe Posix -> DatePicker -> DatePicker
+openPicker settings baseTime pickedTime (DatePicker model) =
     let
-        pickerOffset =
-            calculatePickerOffset zone baseTime pickedTime
+        viewOffset =
+            Utilities.calculateViewOffset settings.zone baseTime pickedTime
+
+        selectionTuple =
+            Maybe.map (\pt -> ( generatePickerDay settings pt, pt )) pickedTime
+
+        timePickerVisible =
+            case settings.timePickerVisibility of
+                NeverVisible ->
+                    False
+
+                Toggleable _ ->
+                    False
+
+                AlwaysVisible _ ->
+                    True
     in
-    DatePicker { model | status = Open baseTime, pickedTime = pickedTime, viewOffset = pickerOffset }
+    DatePicker { model | status = Open timePickerVisible (generatePickerDay settings baseTime), selectionTuple = selectionTuple, viewOffset = viewOffset }
```

Comment:
@Tranquility The `timePickerVisibility` set in the settings is technically separate from the actual state of the time picker visibility. This isn't apparent for `NeverVisible` and `AlwaysVisible` which always map to `False` and `True`, respectively. But the `Toggleable ...` setting alone is not a sufficient indicator of whether or not the picker is open, it merely indicates it can be toggled.

To pass in the visibility setting itself would require us to move the open/closed state onto the `Toggleable` setting. This is not ideal because it a) would no longer be a static setting, b) exposes the time picker state directly to the user to manipulate instead of keeping it internal to the module.